.github/workflows/static-analysis.yml

```diff
@@ -0,0 +1,44 @@
+
+name: "Static Analysis"
+
+on:
+  pull_request:
+    branches:
+      - "*.x"
+  push:
+    branches:
+      - "*.x"
+
+jobs:
+  static-analysis-phpstan:
+    name: "Static Analysis with PHPStan"
+    runs-on: "ubuntu-20.04"
+
+    strategy:
+      matrix:
+        php-version:
+          - "7.4"
+
+    steps:
+      - name: "Checkout code"
+        uses: "actions/checkout@v2"
+
+      - name: "Install PHP"
+        uses: "shivammathur/setup-php@v2"
+        with:
+          coverage: "none"
+          php-version: "${{ matrix.php-version }}"
+          tools: "cs2pr"
+
+      - name: "Cache dependencies installed with composer"
+        uses: "actions/cache@v2"
+        with:
+          path: "~/.composer/cache"
+          key: "php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock') }}"
+          restore-keys: "php-${{ matrix.php-version }}-composer-locked-"
+
+      - name: "Install dependencies with composer"
+        run: "composer install --no-interaction --no-progress --no-suggest"
+
+      - name: "Run a static analysis with phpstan/phpstan"
+        run: "vendor/bin/phpstan analyse --error-format=checkstyle | cs2pr"
```

Comment:
you should just use `vendor/bin/phpstan analyse`. It has native support for github actions now, outputting annotations when needed

Reply:
@stof it seems to work when there are no errors, but otherwise I get `Error: Start tag expected, '<' not found on line 2, column 2`

Reply:
I reverted to the previous solution.

Reply:
Well, you must remove the `| cs2pr` from the command (it is not included in my comment), as `cs2pr` is about converting checkstyle output to github actions annotations, and so makes no sense when the tool outputs annotations directly.

Reply:
Oh okay, I feel dumb now :facepalm: